docs/src/markdown/extensions/magiclink.md

```diff
@@ -479,6 +480,51 @@ and will copy them for your custom repository. If this is not sufficient, you ca
 `shortener_user_exclude` for your custom repository provider using your specified `name`. If you manually set excludes
 in this manner, no excludes from the same `type` will be copied over.
 
+## Simple User-Defined References
+
+You can integrate with additional services like ticketing systems or shortlink providers (sometimes called `go/` links) by defining simple patterns that will be autolinked.
+
+The `simplerefs` configuration supports matching a static prefix followed by an alphanumeric identifier. We look for patterns of form `{ref_prefix}{identifier}` with the following restrictions:
+
+- `ref_prefix` is a configured literal value consisting of a letter followed by alphanumeric characters or punctuation from the set `_/-`; it will usually end with punctuation like `TICKET-`
+- `identifier` is an alphanumeric string containing at least one character; it will be captured and injected into a configured target URL
+- The pattern must be preceeded by whitespace or be at the beginning of the line
+- The pattern must be followed by whitespace or punctuation from the set `.,!?:`
+
+Suppose we have a Jira project called `TICKET` where the issues are referred to like `TICKET-123` and a shortlink service where labels look like `go/myproject`. Our configuration will be:
+
+```
+'simplerefs': [
+    {
+        # Anchor tags will have class name "magiclink-simplerefs-ticket"
+        'ref_prefix': 'TICKET-',
+        # The target must contain "<id>" to indicate where to inject the captured identifier
+        'target_url': 'https://ticket.test.com/TICKET-<id>'
+    },
+    {
+        # Anchor tags will have class name "magiclink-simplerefs-go"
+        'ref_prefix': 'go/',
+        'target_url': 'https://go.test.com/<id>'
+    }
+]
+```
+
+Each `ref_prefix` is paired with a `target_url` that must contain an `<id>` placeholder where the captured identifier will be injected. Although matching is case-insensitive, `simplerefs` will preserve casing of the input text. Note that the `ref_prefix` is also used to derive a class name attached to the links it produces; the class name will be the value of `ref_prefix` downcased and with punctuation removed.
```

Comment:
I don't think all punctuation should be removed. `-` and `_` can remain, but trailing `-` or `_` can probably be stripped from the prefix. If we allowed `/`, we should likely convert it to `-`.

I imagine if we allowed `some/ref`, that both `some` and `ref` should hold to the prefix restriction (start with letter and followed by alphanumeric plus `-` and `_`).